src/components/SettlementButton/index.tsx

```diff
@@ -270,48 +363,193 @@ function SettlementButton({
+            return;
+        }
+
+        if (policy && shouldRestrictUserBillableActions(policy.id)) {
+            Navigation.navigate(ROUTES.RESTRICTED_ACTION.getRoute(policy.id));
+            return;
+        }
+
+        let paymentType;
+        switch (paymentMethod) {
+            case CONST.PAYMENT_METHODS.PERSONAL_BANK_ACCOUNT:
+                paymentType = CONST.IOU.PAYMENT_TYPE.EXPENSIFY;
+                break;
+            case CONST.PAYMENT_METHODS.BUSINESS_BANK_ACCOUNT:
+                paymentType = CONST.IOU.PAYMENT_TYPE.VBBA;
+                break;
+            default:
+                paymentType = CONST.IOU.PAYMENT_TYPE.ELSEWHERE;
+        }
+        triggerKYCFlow(event, paymentType, paymentMethod, selectedPolicy ?? lastPaymentPolicy);
+        if (paymentType === CONST.IOU.PAYMENT_TYPE.EXPENSIFY || paymentType === CONST.IOU.PAYMENT_TYPE.VBBA) {
+            setPersonalBankAccountContinueKYCOnSuccess(ROUTES.ENABLE_PAYMENTS);
+        }
     };
 
+    const getCustomText = () => {
+        if (shouldUseShortForm) {
+            return translate('iou.pay');
+        }
+
+        if (lastPaymentMethod === CONST.IOU.PAYMENT_TYPE.ELSEWHERE) {
+            return translate('iou.payElsewhere', {formattedAmount});
+        }
+
+        return translate('iou.settlePayment', {formattedAmount});
+    };
+
+    const getSecondLineText = (): string | undefined => {
+        if (
+            shouldUseShortForm ||
+            lastPaymentMethod === CONST.IOU.PAYMENT_TYPE.ELSEWHERE ||
+            (paymentButtonOptions.length === 1 && paymentButtonOptions.every((option) => option.value === CONST.IOU.PAYMENT_TYPE.ELSEWHERE)) ||
+            (shouldHidePaymentOptions && (shouldShowApproveButton || onlyShowPayElsewhere))
+        ) {
+            return undefined;
+        }
+
+        if (lastPaymentPolicy) {
+            return lastPaymentPolicy.name;
+        }
+
+        if (lastPaymentMethod === CONST.IOU.PAYMENT_TYPE.EXPENSIFY || (hasIntentToPay && isInvoiceReportUtil(iouReport))) {
+            const bankAccountToDisplay = hasIntentToPay ? (formattedPaymentMethods.at(0) as BankAccount) : bankAccount;
+
+            if (isBusinessInvoiceRoom(chatReport) && bankAccountToDisplay) {
+                return translate('iou.invoiceBussinessBank', {lastFour: bankAccountToDisplay?.accountData?.accountNumber?.slice(-4) ?? ''});
+            }
+
+            if (isIndividualInvoiceRoomUtil(chatReport) && bankAccountToDisplay) {
+                return translate('iou.invoicePersonalBank', {lastFour: bankAccountToDisplay?.accountData?.accountNumber?.slice(-4) ?? ''});
+            }
+
+            if (personalBankAccountList.length === 1 && !hasActivatedWallet) {
+                return translate('paymentMethodList.bankAccountLastFour', {lastFour: personalBankAccountList.at(0)?.accountData?.accountNumber?.slice(-4) ?? ''});
+            }
```

Comment:
I've just found a bug within the invoice flow and I think we don't need all of this condition. We only need to check whether the bankAccount is business or not to decide the message here:
```suggestion
            if (bankAccountToDisplay) {
                return translate(bankAccountToDisplay.accountData?.type === CONST.BANK_ACCOUNT.TYPE.BUSINESS ? 'iou.invoiceBussinessBank' : 'iou.invoicePersonalBank', {lastFour: bankAccountToDisplay?.accountData?.accountNumber?.slice(-4) ?? ''});
            }
```

Reply:
I think for Invoice we don't show `Wallet` 🤔

Reply:
yes but the parent condition has `lastPaymentMethod === CONST.IOU.PAYMENT_TYPE.EXPENSIFY`

Reply:
```diff --git a/src/components/SettlementButton/index.tsx b/src/components/SettlementButton/index.tsx
index 1acd1f878f1..6dbc7f2cb50 100644
--- a/src/components/SettlementButton/index.tsx
+++ b/src/components/SettlementButton/index.tsx
@@ -451,12 +451,8 @@ function SettlementButton({
         if (lastPaymentMethod === CONST.IOU.PAYMENT_TYPE.EXPENSIFY || (hasIntentToPay && isInvoiceReportUtil(iouReport))) {
             const bankAccountToDisplay = hasIntentToPay ? (formattedPaymentMethods.at(0) as BankAccount) : bankAccount;
 
-            if (isBusinessInvoiceRoom(chatReport) && bankAccountToDisplay) {
-                return translate('iou.invoiceBussinessBank', {lastFour: bankAccountToDisplay?.accountData?.accountNumber?.slice(-4) ?? ''});
-            }
-
-            if (isIndividualInvoiceRoomUtil(chatReport) && bankAccountToDisplay) {
-                return translate('iou.invoicePersonalBank', {lastFour: bankAccountToDisplay?.accountData?.accountNumber?.slice(-4) ?? ''});
+            if (bankAccountToDisplay && isInvoiceReportUtil(iouReport)) {
+                return translate('iou.invoicePersonalBank', { lastFour: bankAccountToDisplay?.accountData?.accountNumber?.slice(-4) ?? '' });
             }
```

does this look good to you?

Reply:
Oh I see. Cool Let's do it

Reply:
The return value in your diff should be:
```javascript
return translate(bankAccountToDisplay.accountData?.type === CONST.BANK_ACCOUNT.TYPE.BUSINESS ? 'iou.invoiceBussinessBank' : 'iou.invoicePersonalBank', {lastFour: bankAccountToDisplay?.accountData?.accountNumber?.slice(-4) ?? ''});

```

Reply:
done!